cmake/CompilerHelper.cmake

```diff
@@ -2,13 +2,11 @@
 # Compiler specific settings
 # =============================================================================
 if(CMAKE_CXX_COMPILER_ID MATCHES "Clang")
-  set(CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS
-      "${CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS} -undefined dynamic_lookup")
-endif()
-
-if(CMAKE_C_COMPILER_ID MATCHES "Clang")
-  set(CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS
-      "${CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS} -undefined dynamic_lookup")
+  set(UNDEFINED_SYMBOLS_IGNORE_FLAG "-undefined dynamic_lookup")
+  string(APPEND CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS " ${UNDEFINED_SYMBOLS_IGNORE_FLAG}")
+  string(APPEND CMAKE_SHARED_LIBRARY_CREATE_C_FLAGS " ${UNDEFINED_SYMBOLS_IGNORE_FLAG}")
+else()
+  set(UNDEFINED_SYMBOLS_IGNORE_FLAG "--unresolved-symbols=ignore-all")
```

Comment:
`UNDEFINED_SYMBOLS_IGNORE_FLAG` in gcc is not used (since it's default anyway).
In mac we add it to `CMAKE_SHARED_LIBRARY_CREATE_CXX_FLAGS` so that both compilers end up with similar behavior.
@pramodk raised concern whether we should only apply this only to the strictly necessary cases (libnrnpython). I agree it would be safer, but in that case I would set GCC flags to behave the same

Reply:
I recall for the MPI dynamic code going to great lengths to avoid undefined symbols. Not certain, but it sounds Iike I could have greatly simplified things if I had known about this way of avoiding them.